azurerm/resource_arm_logic_app_workflow.go

```diff
@@ -0,0 +1,264 @@
+				"$schema":        workflowSchema,
+				"contentVersion": workflowVersion,
+				"actions":        make(map[string]interface{}, 0),
+				"triggers":       make(map[string]interface{}, 0),
+			},
+			Parameters: parameters,
+		},
+		Tags: expandTags(tags),
+	}
+
+	_, err := client.CreateOrUpdate(ctx, resourceGroup, name, properties)
+	if err != nil {
+		return fmt.Errorf("[ERROR] Error creating Logic App Workflow %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	read, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("[ERROR] Error making Read request on Logic App Workflow %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if read.ID == nil {
+		return fmt.Errorf("[ERROR] Cannot read Logic App Workflow %q (Resource Group %q) ID", name, resourceGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmLogicAppWorkflowRead(d, meta)
+}
+
+func resourceArmLogicAppWorkflowUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).logicWorkflowsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resourceGroup := id.ResourceGroup
+	name := id.Path["workflows"]
+
+	// lock to prevent against Actions, Parameters or Triggers conflicting
+	azureRMLockByName(name, logicAppResourceName)
+	defer azureRMUnlockByName(name, logicAppResourceName)
+
+	read, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(read.Response) {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("[ERROR] Error making Read request on Logic App Workflow %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if read.WorkflowProperties == nil {
+		return fmt.Errorf("[ERROR] Error parsing Logic App Workflow - `WorkflowProperties` is nil")
+	}
+
+	location := azureRMNormalizeLocation(d.Get("location").(string))
+	parameters := expandLogicAppWorkflowParameters(d)
+	tags := d.Get("tags").(map[string]interface{})
+
+	properties := logic.Workflow{
+		Location: utils.String(location),
+		WorkflowProperties: &logic.WorkflowProperties{
+			Definition: read.WorkflowProperties.Definition,
+			Parameters: parameters,
+		},
+		Tags: expandTags(tags),
+	}
+
+	_, err = client.CreateOrUpdate(ctx, resourceGroup, name, properties)
+	if err != nil {
+		return fmt.Errorf("Error updating Logic App Workspace %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	return resourceArmLogicAppWorkflowRead(d, meta)
+}
+
+func resourceArmLogicAppWorkflowRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).logicWorkflowsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["workflows"]
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
```

Comment:
Same issue here where we should provide logging message when removing resource from state.